### PR TITLE
Vec 381 detailed index list

### PR DESCRIPTION
**Pull request by dwelch-spike**
NOTE: I have this PR pointing at vec-380 because it is based on this branch, hopefully that makes reviewing both the PRs easier.

This PR adds index size and percent unmerged to index ls output, here is a preview.
```
╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                    Indexes                                                   │
├───┬───────┬───────────┬────────┬────────────┬─────────────────┬──────────┬────────────────┬──────┬───────────┤
│   │ NAME  │ NAMESPACE │ FIELD  │ DIMENSIONS │ DISTANCE METRIC │ UNMERGED │ VECTOR RECORDS │ SIZE │ UMERGED % │
├───┼───────┼───────────┼────────┼────────────┼─────────────────┼──────────┼────────────────┼──────┼───────────┤
│ 1 │ idxfy │ test      │ vector │        256 │ COSINE          │        0 │              0 │ 0 B  │ 0%        │
╰───┴───────┴───────────┴────────┴────────────┴─────────────────┴──────────┴────────────────┴──────┴───────────╯
```

### CODE REVIEW

**Comment by dwelch-spike** [comment]
Note: Adam wanted me to remove "inf" and "inifinity" as possible cache arguments so I stuck that in here too.

File: cmd/flags/optionals.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (f *InfDurationOptionalFlag) Set(val string) error {
 
 	val = strings.ToLower(val)
 
-	if val == "inf" || val == "infinity" || val == "-1" {
+	if val == "-1" {
```
Comment (reugn) [inline cmd/flags/optionals.go:227]: Consider using the `flags.Infinity` constant.

File: cmd/flags/optionals.go [hunk 2]
```diff
@@ -210,7 +210,7 @@ func (f *DurationOptionalFlag) Int64() *int64 {
 }
 
 // InfDurationOptionalFlag is a flag that can be either a time.duration or infinity.
-// It is used for flags like --hnsw-index-cache-expiry which can be set to "infinity"
+// It is used for flags like --hnsw-index-cache-expiry which can be set to infinity (-1)
```
Comment (reugn) [inline cmd/flags/optionals.go:213]: Consider replacing "infinity" with "never expire".

**Review by reugn — APPROVED** [review]

File: cmd/flags/optionals.go (hunk 2 above)
Comment (dwelch-spike) [reply cmd/flags/optionals.go]: Changed.

**Pull request merged**